backend/api/views.py

```
@@ -39,32 +47,35 @@
     serializer_class = PaymentSerializer
+    permission_classes = [IsAuthenticated]
     
-    # Removed permission_classes
     def get_queryset(self):
+        # Users can only see their own payments
         return Payment.objects.filter(user=self.request.user)
 
 class CoinTransactionViewSet(viewsets.ModelViewSet):
     queryset = CoinTransaction.objects.all()
     serializer_class = CoinTransactionSerializer
+    permission_classes = [IsAuthenticated]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)